kedro/framework/cli/starters.py

```diff
@@ -631,21 +635,46 @@ def _fetch_config_from_file(config_path: str) -> dict[str, str]:
     """
     try:
         with open(config_path, encoding="utf-8") as config_file:
-            config = yaml.safe_load(config_file)
+            config_input = yaml.safe_load(config_file)
 
         if KedroCliError.VERBOSE_ERROR:
             click.echo(config_path + ":")
-            click.echo(yaml.dump(config, default_flow_style=False))
+            click.echo(yaml.dump(config_input, default_flow_style=False))
     except Exception as exc:
         raise KedroCliError(
             f"Failed to generate project: could not load config at {config_path}."
         ) from exc
 
-    # The return type defined is more specific than the "Any" type config return from yaml.safe_load
-    return config  # type: ignore[no-any-return]
+    config = (
+        {str(key): str(val) for key, val in config_input.items()}
+        if config_input
+        else {}
+    )
+
+    if starter_alias and ("tools" in config or "example_pipeline" in config):
+        raise KedroCliError(
+            "The --starter flag can not be used with `example_pipeline` and/or `tools` keys in the config file."
```

Comment:
Is this check not redundant as `_validate_flag_inputs` already checks for this?

Reply:
> Is this check not redundant as `_validate_flag_inputs` already checks for this?

Looks like _fetch_validate_parse_config_from_file() checks input from config file, and _validate_flag_inputs() validates inout from CLI?

Reply:
Absolutely correct! I got my wires crossed